### PR TITLE
Fix: Display preferred name if user has one

### PR DESCRIPTION
**Pull request by ellen-wright**
Closes #UserOfficeProject/user-office-project-issue-tracker/issues/766

## Description
Uses the user's preferred name if they have one, otherwise still uses their first name.  

## Motivation and Context
Requested in user feedback. 

## How Has This Been Tested
Manually tested on users with and without a set preferred name. 

## Changes
STFC Data source initially looks for a preferred name if their isn't one the preferredname field is set to the user's first name. 
When viewing as user preferred name is used over first name if there is one. 

## Tests included/Docs Updated?

- [ ] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Review by mutambaraf — COMMENTED** [review]
It would me nice if we can test this feature .

**Comment by bolmsten** [comment]
Add E2E, one with a user that has a preferred name and one without

File: apps/frontend/src/hooks/user/useUserData.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ export function useBasicUserData(userId?: number) {
 export interface BasicUserData {
   firstname: string;
   lastname: string;
-  preferredname: string;
   organisation: string;
   id: number;
+  preferredname: string | undefined | null;
```
Comment (mutambaraf) [inline apps/frontend/src/hooks/user/useUserData.ts:118]: You can just make this optional.

**Comment by ellen-wright** [comment]
Blocked by issue with mock-uows

File: apps/frontend/src/utils/user.ts [hunk 2]
```diff
@@ -5,10 +5,17 @@ export const getFullUserName = (
 ): string => (user ? `${user.firstname} ${user.lastname}` : 'None');
```
Comment (martin-trajanovski) [inline apps/frontend/src/utils/user.ts:5]: I was just wondering if we need to do the same `preferredname` logic here as well. I see that this function `getFullUserName` is quite similar to the `getFullUserNameWithEmail` but it doesn't use the preferred name logic. Maybe I am wrong but just wanted to make it clear.

File: apps/frontend/src/components/proposal/ProposalParticipant.tsx [hunk 3]
```diff
@@ -55,7 +55,9 @@ export default function ProposalParticipant(props: {
           label="Principal Investigator"
           value={
             props.principalInvestigator
-              ? `${props.principalInvestigator.firstname} ${props.principalInvestigator.lastname}; ${props.principalInvestigator.organisation}`
+              ? `${props.principalInvestigator.preferredname}`
```
Comment (martin-trajanovski) [inline apps/frontend/src/components/proposal/ProposalParticipant.tsx:58]: Maybe it is good idea to extract this in a function to be more clear what is it about. For example `getFullUserNameWithOrganisation` something similar as `getFullUserNameWithEmail`.

**Review by martin-trajanovski — COMMENTED** [review]
In general it looks good to me. I just left two small comments that could be checked.

**Review by mutambaraf — APPROVED** [review]

**Review by martin-trajanovski — APPROVED** [review]
It looks good to me

**Pull request merged**